### PR TITLE
feat: add dds instances data source

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add dds instances data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add dds instances data source
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dds/' TESTARGS='-run TestAccDatasourceDdsInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dds/ -v -run TestAccDatasourceDdsInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceDdsInstance_basic
=== PAUSE TestAccDatasourceDdsInstance_basic
=== CONT  TestAccDatasourceDdsInstance_basic
--- PASS: TestAccDatasourceDdsInstance_basic (787.80s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dds       787.848s
```

### CODE REVIEW

File: huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+		},
+	}
+	return &sc
+}
+
+func resourceBackupRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+
+	var mErr *multierror.Error
```
Comment (deer-hang) [inline huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go:246]: remove this line to line:223

Comment (houpeng80) [reply huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go:246]: the code is automatically generated, the style is unified

File: huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+				Type:        schema.TypeList,
+				Elem:        BackupInstanceSchema(),
+				Computed:    true,
+				Description: `Indicates the list of DDS instances.`,
+			},
+		},
+	}
+}
+
+func BackupInstanceSchema() *schema.Resource {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go]: All of **Backup** s should be updated to **Dds**

File: docs/data-sources/dds_instances.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+* `db_username` - Indicates the DB Administator name.
+
+* `status` - Indicates the the DB instance status.
+
+* `enterprise_project_id` - Indicates the enterprise project id of the dds instance.
+
+* `nodes` - Indicates the instance nodes information.
+  The [Node](#DdsInstance_InstanceNode) structure is documented below.
+
+* `tags` - Indicates The key/value pairs to associate with the DDS instance.
```
Comment (ShiChangkuo) [inline docs/data-sources/dds_instances.md]:
```suggestion
* `tags` - Indicates the key/value pairs to associate with the DDS instance.

```

File: huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+			},
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the VPC ID`,
+			},
+			"subnet_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the VPC ID.`,
```
Comment (ShiChangkuo) [inline huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go]: the description is  not correct.

File: huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+	var mErr *multierror.Error
+
+	// getDDSInstances: Query the List of DDS instances.
+	var (
+		getDDSInstancesHttpUrl = "v3/{project_id}/instances"
+		getDDSInstancesProduct = "dds"
+	)
+	getDDSInstancesClient, err := config.NewServiceClient(getDDSInstancesProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating Backup Client: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go]: the error message is not correct

File: huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+	getDDSInstancesPath += getDDSInstancesQueryParams
+
+	getDDSInstancesResp, err := pagination.ListAllItems(
+		getDDSInstancesClient,
+		"offset",
+		getDDSInstancesPath,
+		&pagination.QueryOpts{MarkerField: ""})
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Backup")
```
Comment (ShiChangkuo) [inline huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go]: ditto

File: huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go]: done

File: docs/data-sources/dds_instances.md (hunk 3 above)
Comment (houpeng80) [reply docs/data-sources/dds_instances.md]: done

File: huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go]: done

File: huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go (hunk 5 above)
Comment (houpeng80) [reply huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go]: done

File: huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go (hunk 6 above)
Comment (houpeng80) [reply huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go]: done

**Pull request merged**